### PR TITLE
[docs] Add content for exporting OpenTelemetry metrics

### PR DESCRIPTION
**Pull request by EamonnTP**
## Summary

This PR updates the current OpenTelemetry integration [documentation](https://www.elastic.co/guide/en/apm/get-started/current/open-telemetry-elastic.html) with the recent support for exporting OpenTelemetry metrics.  To accomodate the new content, I have restructured the content flow and included new verifying and visualization data sections. 

**Still to do** - add a screenshot of the dashboard. 

## Docs preview

https://apm-server_4498.docs-preview.app.elstc.co/guide/en/apm/get-started/master/open-telemetry-elastic.html

## Related issues

https://github.com/elastic/apm/issues/337
Closes https://github.com/elastic/apm/issues/363

### CODE REVIEW

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 1]
```diff
@@ -92,63 +96,159 @@ You can also build the collector-contrib repository by cloning it and running:
...
+* Field: `system.cpu.load_average.1m`.
+* Group by: `Terms`.
+* By: `host.ip`.
+* Top: `10`.
+* Order by: `Doc Count (default)`.
+* Direction: `Descending`.
+. Click *Save and return*.
++
+Both visualizations are now displayed on your custom dashboard.
+
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:251]: I'll be curious to see how difficult this is to maintain.

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 2]
```diff
@@ -92,63 +96,159 @@ You can also build the collector-contrib repository by cloning it and running:
...
+      traces:
+        receivers:
+          - otlp
+        processors:
+          - batch
+        exporters:
+          - elastic <4>
 ----
+<1> The `hostmetrics` receiver must be defined to generate metrics about the host system scraped from various sources.
+<2> At a minimum, you must define the URL of the APM Server instance you are sending data to. See the <<open-telemetry-elastic-config-ref,configuration reference>>
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:180]: Matching the other callouts

```suggestion
<2> At a minimum, you must define the URL of the APM Server instance you are sending data to. See the <<open-telemetry-elastic-config-ref,configuration reference>>.
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 3]
```diff
@@ -28,10 +32,11 @@ If you're unfamiliar with the project, see the {ot-spec}[spec] for more informat
...
-This exporter translates the OpenTelemetry trace data collected from your services to Elastic's protocol,
-before sending the data to the Elastic Stack.
+
+Before sending the data to the {stack}, the exporter translates the OpenTelemetry trace data collected from your services
+and the metrics data collected from your infrastructure and applications to Elastic's protocol.
 By extending the OpenTelemetry collector,
-no changes are needed in your instrumented services in order to begin using the Elastic Stack.
+no changes are needed in your instrumented services to begin using the {stack}.
 
 [role="screenshot"]
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:41]: Let's remove the screenshot css from this.

```suggestion
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 4]
```diff
@@ -28,10 +32,11 @@ If you're unfamiliar with the project, see the {ot-spec}[spec] for more informat
...
 image::images/open-telemetry-elastic-arch.png[OpenTelemetry Elastic architecture diagram]
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:42]: Should we update the diagram to include metrics?

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 5]
```diff
@@ -40,20 +45,19 @@ image::images/open-telemetry-elastic-arch.png[OpenTelemetry Elastic architecture
...
+* `processors`: Defines optional transformations that occur between receiving and exporting data.
+* `exporters`: Configures how data is sent to its destination--in this case, the {stack}.
 
-Once a `receiver`, `processor`, and `exporter` is defined, `pipelines` can be configured in the `services` section of your configuration. Specifically, a `traces` pipeline will define the path of trace data through your collector, and bring all three of these components together.
+Once a `receiver`, `processor`, and `exporter` is defined, `pipelines` can be configured in the `services` section of your configuration.
+The `traces` and `metrics` pipelines define the path of trace data and metrics through your collector and bring all three of these components together.
 
-TIP: More information is available in the
-{ot-pipelines}[OpenTelemetry pipeline docs]
+TIP: More information is available in the {ot-pipelines}[OpenTelemetry pipeline docs].
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:57]: OTel got rid of their pipeline docs and this link is now a 404. We should update it on line six. Maybe this? https://github.com/open-telemetry/opentelemetry-collector/blob/master/processor/README.md

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 6]
```diff
@@ -69,7 +73,7 @@ OpenTelemetry Collectors can be run as an Agent, or as standalone collectors.
 They can be deployed as often as necessary and scaled up or out.
 
 Deployment planning resources are available in OpenTelemetry's {ot-collector}[Getting Started]
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:75]: This link is also a 404 and should be updated

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 7]
```diff
@@ -92,63 +96,159 @@ You can also build the collector-contrib repository by cloning it and running:
...
-[float]
-[[open-telemetry-elastic-configure]]
-==== Configure the collector
+[[open-telemetry-elastic-traces-metrics]]
+==== Collect traces and metrics
 
-Create a `yaml` configuration file.
+To export traces and metrics to the OpenTelemetry Collector, ensure that you have instrumented your services and applications
+with the OpenTelemetry API and/or SDK. If you haven't, see the Elastic APM <<install-and-run,install and run guide>> to get started
+with Elastic APM Agents instead.
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:104]: I like the examples below, but I think we should be even more clear here that the main purpose of this guide is OTel integration with Elastic, not OTel agent setup. What if we added a NOTE to the top of this section, and a link to OTel api/sdk setup at the bottom?

---

NOTE: This guide assumes your services and applications have already been instrumented with the OpenTelemetry API and/or SDK. If you're new to APM, we recommend <<install-and-run,getting started with Elastic APM agents>> instead.

To export traces and metrics to the OpenTelemetry Collector, ensure that you have instrumented your services and applications with the OpenTelemetry API and/or SDK.

Here is an example of how to set up the OpenTelemetry Java agent.

// example

Here is an example of how to capture business metrics from an application.

// example

For more information on setting up OTel, see <link>.

---

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 8]
```diff
@@ -92,63 +96,159 @@ You can also build the collector-contrib repository by cloning it and running:
...
+[[open-telemetry-elastic-caveats]]
+==== Caveats
+
+If collecting metrics, please note that the https://www.javadoc.io/doc/io.opentelemetry/opentelemetry-api/latest/io/opentelemetry/api/metrics/DoubleValueRecorder.html[`DoubleValueRecorder`]
+and https://www.javadoc.io/doc/io.opentelemetry/opentelemetry-api/latest/io/opentelemetry/api/metrics/LongValueObserver.html[`LongValueRecorder`] metrics are not yet supported.
+
+[[open-telemetry-elastic-verify]]
+==== Verify OpenTelemetry metrics data
+
+To validate that metrics are successfully being reported to {kib} is by using *Discover*.
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:198]:
```suggestion
Use *Discover* to validate that metrics are successfully being reported to {kib}.
```

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 9]
```diff
@@ -92,63 +96,159 @@ You can also build the collector-contrib repository by cloning it and running:
...
+. Select `apm-*` as your index pattern.
+. Filter the data to only show documents with metrics: `processor.name :"metric"`
+. Narrow your search with a known OpenTelemetry field. For example, if you have an `order_value` field, add `order_value: *` to your search to return
+only OpenTelemetry metrics documents.
+
+[[open-telemetry-elastic-kibana]]
+==== Visualize in {kib}
+
+TSVB within {kib} is the recommended visualization for OpenTelemtry metrics. TSVB is a time series data visualizer that allows you to use the full power of the
+{ess} aggregation framework. With TSVB, you can combine an infinite number of aggregations to display complex data.
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:216]: Is `{ess}` correct here?

**Review by bmorelli25 — COMMENTED** [review]
Ran out of time today to spin this up and actually test it. Will do that part tomorrow!

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 4 above)
Comment (axw) [reply docs/guide/opentelemetry-elastic.asciidoc:42]: I don't think so. The diagram doesn't mention data types, and there's no change to the deployment architecture.

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 5 above)
Comment (axw) [reply docs/guide/opentelemetry-elastic.asciidoc:57]: or https://opentelemetry.io/docs/collector/configuration/ ?

**Review by axw — COMMENTED** [review]
Looking good!

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 2 above)
Comment (EamonnTP) [reply docs/guide/opentelemetry-elastic.asciidoc:180]: The end of the sentence is on the next line.

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 3 above)
Comment (EamonnTP) [reply docs/guide/opentelemetry-elastic.asciidoc:41]: Will do. Thanks.

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 5 above)
Comment (EamonnTP) [reply docs/guide/opentelemetry-elastic.asciidoc:57]: Thanks. I added https://opentelemetry.io/docs/collector/configuration/

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 6 above)
Comment (EamonnTP) [reply docs/guide/opentelemetry-elastic.asciidoc:75]: Thanks. Added https://opentelemetry.io/docs/collector/getting-started/

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 7 above)
Comment (EamonnTP) [reply docs/guide/opentelemetry-elastic.asciidoc:104]: Great suggestion. Thanks!

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 8 above)
Comment (EamonnTP) [reply docs/guide/opentelemetry-elastic.asciidoc:198]: Thanks.

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 9 above)
Comment (EamonnTP) [reply docs/guide/opentelemetry-elastic.asciidoc:216]: Ah! Should be {es} Thanks.

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 10]
```diff
@@ -92,63 +96,159 @@ You can also build the collector-contrib repository by cloning it and running:
...
+. In the side navigation, under *{kib}*, select *Discover*.
+. Select `apm-*` as your index pattern.
+. Filter the data to only show documents with metrics: `processor.name :"metric"`
+. Narrow your search with a known OpenTelemetry field. For example, if you have an `order_value` field, add `order_value: *` to your search to return
+only OpenTelemetry metrics documents.
+
+[[open-telemetry-elastic-kibana]]
+==== Visualize in {kib}
+
+TSVB within {kib} is the recommended visualization for OpenTelemtry metrics. TSVB is a time series data visualizer that allows you to use the full power of the
```
[the remark this answers is not in the document]
Comment (EamonnTP) [reply docs/guide/opentelemetry-elastic.asciidoc:215]: Oops. Thanks!

**Comment by EamonnTP** [comment]
jenkins run the tests please

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 11]
```diff
@@ -92,63 +95,168 @@ You can also build the collector-contrib repository by cloning it and running:
...
+[role="screenshot"]
+image::images/ecommerce-dashboard.png[OpenTelemetry visualizations]
+
+Let's have a look at how this dashboard was created, specifically the Sales USD and System load visualizations.
+
+. In the side navigation, under *{kib}*, select *Dashboard*.
+. Click *Dashboard* and select *Create dashboard*.
+. Click *Save*, enter the name of your dashboard, and then click *Save* again.
+. Let’s add a Sales USD visualization. Click *Edit*.
+. Click *Create new* and then select *TSVB*.
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:234]: My flow was a little different. What version were you using?

. In the side navigation, under *{kib}*, select *Dashboard*.
. Select *Create new dashboard*.
. Click *Save*, enter the title of your dashboard, and then click *Save* again.
. Let’s add a Sales USD visualization. Click *Edit*.
. Click *Create new* and then select *TSVB*.

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 12]
```diff
@@ -92,63 +95,168 @@ You can also build the collector-contrib repository by cloning it and running:
...
+. For the label name, enter Sales USD, and then select the following:
++
+* Aggregation: `Positive Rate`.
+* Field: `order_sum`.
+* Scale: `auto`.
+* Group by: `Everything`
+. Click *Save and return*.
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:241]: My button was only titled "Save". I also had to title the viz at this point. Only after that could I click Save and return

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 2 above)
Comment (bmorelli25) [reply docs/guide/opentelemetry-elastic.asciidoc:184]: lol. Oops!

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 13]
```diff
@@ -92,63 +95,168 @@ You can also build the collector-contrib repository by cloning it and running:
...
+. Enter the label name, System load per host IP, and then select the following:
++
+* Aggregation: `Average`.
+* Field: `system.cpu.load_average.1m`.
+* Group by: `Terms`.
+* By: `host.ip`.
+* Top: `10`.
+* Order by: `Doc Count (default)`.
+* Direction: `Descending`.
+. Click *Save and return*.
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:253]: Same as above

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 14]
```diff
@@ -92,63 +95,168 @@ You can also build the collector-contrib repository by cloning it and running:
...
+   ...
+   // increment business metrics for monitoring
+   orderValueCounter.add(orderPrice);
+}
 ----
 
-If we put everything together, here's an example configuration file that accepts input from an OpenTelemetry Agent,
-processes the data, and sends it to an {ess} instance.
+For more information on setting up OpenTelemetry, see the OpenTelemetry's {ot-collector}[Getting Started]
+documentation.
```
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:135]: I would change this to either this:
```diff
- For more information on setting up OpenTelemetry, see the OpenTelemetry's {ot-collector}[Getting Started]
+ For more information on setting up OpenTelemetry, see the OpenTelemetry {ot-collector}[Getting Started]
```

Or this:
```diff
- For more information on setting up OpenTelemetry, see the OpenTelemetry's {ot-collector}[Getting Started]
+ For more information on setting up OpenTelemetry, see OpenTelemetry's {ot-collector}[Getting Started]
```
For more information on setting up OpenTelemetry, see the OpenTelemetry's {ot-collector}[Getting Started]
documentation.

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 11 above)
Comment (EamonnTP) [reply docs/guide/opentelemetry-elastic.asciidoc:234]: The difference is that I found the longest possible route to creating dashboards :-) Followed your steps and they worked perfectly. I'll update the steps.

**Review by bmorelli25 — APPROVED** [review]
Pending Cyrille's input on https://github.com/elastic/apm-server/pull/4498#discussion_r539031821, this LGTM.

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 15]
```diff
@@ -28,10 +32,11 @@ If you're unfamiliar with the project, see the {ot-spec}[spec] for more informat
 
 Elastic's integration is designed to drop into your current OpenTelemetry setup.
 We've done this by extending the "contrib" OpenTelemetry collector and adding an Elastic exporter.
-This exporter translates the OpenTelemetry trace data collected from your services to Elastic's protocol,
-before sending the data to the Elastic Stack.
+
+Before sending the data to the {stack}, the exporter translates the OpenTelemetry trace data collected from your services
+and the metrics data collected from your infrastructure and applications to Elastic's protocol.
```
[the remark this answers is not in the document]
Comment (cyrille-leclerc) [reply docs/guide/opentelemetry-elastic.asciidoc:37]: @axw my challenge is that I also want to provide guidance to people who start adopting OpenTelemetry to collect infrastructure metrics. LEt's see how we can simultaneously recommend to use metricbeat for infrastructure metrics and make successful people deciding to use OpenTelemetry to collect infrastructure metrics.

File: docs/guide/opentelemetry-elastic.asciidoc [hunk 16]
```diff
@@ -92,63 +95,167 @@ You can also build the collector-contrib repository by cloning it and running:
 make otelcontribcol
 ----
```
Comment (cyrille-leclerc) [inline docs/guide/opentelemetry-elastic.asciidoc:96]: @axw @EamonnTP I propose to remove the explanation on how to build yourself the collector. The collector is stable enough and official releases cover the feature we need.
Building your own Otel Collector is now an advanced use case, it's not the purpose of this documentation page.

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 15 above)
[the remark this answers is not in the document]
Comment (cyrille-leclerc) [reply docs/guide/opentelemetry-elastic.asciidoc:37]: I have a proposal:
* We keep the reference to infrastructure metrics switching infrastructure and application: "...and the metrics data collected from your applications and infrastructure..."
* And we add a `TIP`: "To collect infrastructure metrics, it is recommended to still use the collector Elastic Metricbeat to get a more mature collector with more integrations and with more integrated visualizations"

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 16 above)
Comment (EamonnTP) [reply docs/guide/opentelemetry-elastic.asciidoc:96]: Thanks, will do.

File: docs/guide/opentelemetry-elastic.asciidoc (hunk 15 above)
[the remark this answers is not in the document]
Comment (EamonnTP) [reply docs/guide/opentelemetry-elastic.asciidoc:37]: Your proposal sounds to me.

**Review by cyrille-leclerc — APPROVED** [review]

**Pull request merged**